addons/product/models/product_product.py

```diff
@@ -26,12 +26,12 @@ class ProductProduct(models.Model):
     # price_extra: catalog extra value only, sum of variant extra attributes
     price_extra = fields.Float(
         'Variant Price Extra', compute='_compute_product_price_extra',
-        digits='Product Price',
+        digits=0, min_digits='Product Price',
         help="This is the sum of the extra price of all attributes")
     # lst_price: catalog value + extra, context dependent (uom)
     lst_price = fields.Float(
         'Sales Price', compute='_compute_product_lst_price',
-        digits='Product Price', inverse='_set_product_lst_price',
+        digits=0, min_digits='Product Price', inverse='_set_product_lst_price',
```

Comment:
Doesn't the precision depend on the product category?
And it can also be different if buying or selling.
For instance in a gas station in Belgium (PoV of the owner):
* buying gas: 4 digits
* selling gas: 3 digits
* selling snacks: 2 digits